### PR TITLE
Update 15_documentation.md

### PR DESCRIPTION
**Pull request by oerc0122**

### CODE REVIEW

File: documentation/smg/15_documentation.md [hunk 1]
```diff
@@ -21,25 +21,19 @@ On Linux (Optional):
...
+* Cmake **(Recommended)**
 
-* Linux (With Make)
-
-  With `make` installed simply run:
-
-      make html
+  Use the appropriate build command to make the docs following standard CMake procedures (e.g. `make docs`). 
+  
+  This automatically determine OS specific commands and will check if the appropriate programs are available and writes the documentation to `documentation/user_docs/build/html` by default. It also performs cleanup of the documentation's temporary sidebar links.
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:30]: determineS

File: documentation/smg/15_documentation.md [hunk 2]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+
+When producing a new version, the `conf.py` (in `documentation/user_docs/docs`) should have the version updated to match that of the version being built. The documentation should be built manually (as described [here](#building-docs))
+
 ## Deploying documentation
 
-The documentation will be deployed automatically on a nightly build to the `latest` page of the GitHub pages. 
-On release, documentation will be deployed to the folder corresponding to the version number of the relase and redirected to by the `stable` page of the GitHub pages
+### Automatic procedure (recommended)
+The documentation will be deployed automatically on a nightly build to the `unstable` page of the GitHub pages (`https://pace-neutrons.github.io/Horace/unstable`). 
+On release, documentation will be deployed to the folder corresponding to the version number of the relase and redirected to by the `stable` page of the GitHub pages.
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:93]: not sure what "redirected to by" means

File: documentation/smg/15_documentation.md [hunk 3]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+
+### Manual procedure
+This should be used only in dire circumstances i.e. if forcing an update to old documentation or Jenkins failure, and not as the main method.
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:96]: i.e. : do you mean for example, so e.g.? There may be other dire circumstances?

File: documentation/smg/15_documentation.md [hunk 4]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+
+First the project should be checked out to the version whose docs need to be updated. 
+
+The docs should be built (see [here](#building-docs)) and the appropriate substitutions made (**N.B.** not necessary if built through CMake):
+
+* Linux
+  
+  From within the built HTML directory (`documentation/user_docs/build/html`) run:
+     
+  `sed -r -i '/\[NULL\]/d' *html`
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:108]: do you need backticks?

File: documentation/smg/15_documentation.md [hunk 5]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+  
+* Windows
+
+  From within the root Horace dir (or set `-Path` appropriately) run:
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:112]: and powershell

File: documentation/smg/15_documentation.md [hunk 6]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+     
+  ```
+  Foreach($f in Get-ChildItem -Path documentation/user_docs/build/html -Filter *.html) { \
+     (Get-Content $f.FullName) | Where-Object {$_ -notmatch '\\[NULL\\]'} | Set-Content $f.FullName \    
+           }
+  ```
+  
+  These should then be copied outside the project (recommend equivalent to \[from Horace root dir\] `version_number='3.5.3'; mv documentation/user_docs/build/html ../${version_number}`
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:120]: not quite clear - you mean do a windows equivalent of this linux command? Easier to specify that command?

File: documentation/smg/15_documentation.md [hunk 7]
```diff
@@ -88,7 +82,43 @@ More info can be found https://www.sphinx-doc.org/en/master/usage/restructuredte
...
+  
+  The next step is to checkout the `gh-pages` (**N.B**, you may need to forcibly ignore changes to the repo from building the docs \[`git reset --hard` will do this, but care must be taken\]), then move the built documentation from outside the project to a folder of the appropriate version name in the project root directory. 
+  
+  If the documentation is the most recent version of the documentation (i.e. `stable`), within the `stable` folder update `index.html` to redirect to the appropriate documentation version folder (the one which has been just copied).
```
Comment (cmarooney-stfc) [inline documentation/smg/15_documentation.md:124]: comma between folder and update

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

**Review by abuts — APPROVED** [review]
This clears-up a lot for me. Looks good

File: documentation/smg/15_documentation.md (hunk 2 above)
Comment (oerc0122) [reply documentation/smg/15_documentation.md:93]: HTML web redirect

File: documentation/smg/15_documentation.md (hunk 4 above)
Comment (oerc0122) [reply documentation/smg/15_documentation.md:108]: Those backticks put it in a monospaced codeblock in the rendered markdown

File: documentation/smg/15_documentation.md (hunk 6 above)
Comment (oerc0122) [reply documentation/smg/15_documentation.md:120]: Or just whatever is a convenient command

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**